In [7]:
# 前回の分析結果を読み込みたい

import pickle

with open("../第4回/分析結果.pkl", "rb") as f:
    result = pickle.load(f)

In [8]:
result.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                    魅力度   R-squared:                       0.595
Model:                            OLS   Adj. R-squared:                  0.567
Method:                 Least Squares   F-statistic:                     21.04
Date:                Tue, 12 Nov 2024   Prob (F-statistic):           1.53e-08
Time:                        11:19:40   Log-Likelihood:                -163.04
No. Observations:                  47   AIC:                             334.1
Df Residuals:                      43   BIC:                             341.5
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const         18.6402      2.117      8.806      0.000      14.371      22.909
人口         -2.304e-06   7.99e-07     -2.883      0.006   -3.92e-06   -6.92e-07
ホテル客室数         0.0008      0.000      5.781      0.000       0.001       0.001
旅館客室数      -3.578e-05      0.000     -0.242      0.810      -0.000       0.000
==============================================================================
Omnibus:                       14.240   Durbin-Watson:                   1.048
Prob(Omnibus):                  0.001   Jarque-Bera (JB):               16.469
Skew:                           1.102   Prob(JB):                     0.000265
Kurtosis:                       4.885   Cond. No.                     6.89e+06
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 6.89e+06. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

# 決定係数

# 1.全体の解釈

In [9]:
# 決定係数
# R-squered 0.595
result.rsquared

0.5948301261418372

In [10]:
# 自由d調整済み決定係数
result.rsquared_adj

0.5665624605238259

# F検定

In [11]:
#　F検定のp値のみの取得
result.f_pvalue


1.5274367685392574e-08

有意水準を **0.05** としてモデル適合度のF検定を行った結果p値は **1.5e-10** であり、全ての偏回帰係数が0であるという帰無仮説が **棄却された** 。すなわち、今回の分析の重回帰モデルは **有意に妥当な変数が含まれている** 事が分かった。

## 2. 偏回帰係数の解釈

In [12]:
result.pvalues

const     3.529716e-11
人口        6.130946e-03
ホテル客室数    7.617077e-07
旅館客室数     8.098594e-01
dtype: float64

In [ ]:
# 検定だと、相関がない

In [13]:
# 係数のチェックをしましょう
result.params

const     18.640181
人口        -0.000002
ホテル客室数     0.000824
旅館客室数     -0.000036
dtype: float64

有意水準を下回った変数の内、偏回帰係数の付合が負となったのは「人口」であり、偏回帰係数の付合が正となったのは「ホテル客室数」であった。
変数「人口」の偏回帰係数は「-0.000002」であり、変数「人口」が1単位増加すると、目的変数「魅力度」は「-0.00002」減少する。
変数＊＊＊の偏回帰係数は＊＊＊であり、変数「ホテル客室数」が1単位増加すると、目的変数「魅力度」0.000824増加する。

# 切片の解釈

In [14]:
result.params

const     18.640181
人口        -0.000002
ホテル客室数     0.000824
旅館客室数     -0.000036
dtype: float64

In [15]:
result.conf_int()

0             1
const   14.371452  2.290891e+01
人口      -0.000004 -6.922082e-07
ホテル客室数   0.000537  1.111527e-03
旅館客室数   -0.000334  2.622713e-04